### PR TITLE
feat(prompt): add prompt table, entity and repository

### PR DESCRIPTION
**Pull request by mamadoudicko**
# Description

- Add prompt table (and new migration)
- Add Prompt entity
- Add prompt repository (get, create, post, delete)
- Add GET /prompts to retrieve all public prompts

Related prompt: https://github.com/StanGirard/quivr/issues/812

### CODE REVIEW

File: backend/core/repository/prompt/create_prompt.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    content: str
+    status: str = "private"
+
+
+def create_prompt(prompt: CreatePromptProperties) -> Prompt:
+    """Create a prompt by id"""
+    commons = common_dependencies()
+
+    response = (
+        commons["supabase"].from_("prompts").insert(prompt.dict()).execute()
```
Comment (StanGirard) [inline backend/core/repository/prompt/create_prompt.py:19]: Nope !

File: backend/core/repository/prompt/delete_prompt_py_id.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    Args:
+        prompt_id (UUID): The id of the prompt
+
+    Returns:
+        Prompt: The prompt
+    """
+    commons = common_dependencies()
+
+    response = (
+        commons["supabase"]
```
Comment (StanGirard) [inline backend/core/repository/prompt/delete_prompt_py_id.py:20]: Nope

File: backend/core/repository/prompt/get_prompt_by_id.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    Args:
+        prompt_id (UUID): The id of the prompt
+
+    Returns:
+        Prompt: The prompt
+    """
+    commons = common_dependencies()
+
+    response = (
+        commons["supabase"]
```
Comment (StanGirard) [inline backend/core/repository/prompt/get_prompt_by_id.py:20]: Nope

File: backend/core/repository/prompt/get_public_prompts.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def get_public_prompts() -> list[Prompt]:
+    """
+    List all public prompts
+    """
+    commons = common_dependencies()
+
+    return (
+        commons["supabase"]
```
Comment (StanGirard) [inline backend/core/repository/prompt/get_public_prompts.py:12]: Nope

File: backend/core/repository/prompt/delete_prompt_py_id.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    Delete a prompt by id
+    Args:
+        prompt_id (UUID): The id of the prompt
+
+    Returns:
+        Prompt: The prompt
+    """
+    commons = common_dependencies()
+
+    response = (
```
[the remark this answers is not in the document]
Comment (mamadoudicko) [reply backend/core/repository/prompt/delete_prompt_py_id.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/repository/prompt/delete_prompt_py_id.py (hunk 2 above)
Comment (mamadoudicko) [reply backend/core/repository/prompt/delete_prompt_py_id.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/repository/prompt/get_prompt_by_id.py (hunk 3 above)
Comment (mamadoudicko) [reply backend/core/repository/prompt/get_prompt_by_id.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/repository/prompt/get_public_prompts.py (hunk 4 above)
Comment (mamadoudicko) [reply backend/core/repository/prompt/get_public_prompts.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/repository/prompt/update_prompt_by_id.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    content: Optional[str]
+    status: Optional[str]
+
+
+def update_prompt_by_id(prompt_id: UUID, prompt: PromptUpdatableProperties) -> Prompt:
+    """Update a prompt by id"""
+    commons = common_dependencies()
+
+    response = (
+        commons["supabase"]
```
[the remark this answers is not in the document]
Comment (mamadoudicko) [reply backend/core/repository/prompt/update_prompt_by_id.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/repository/prompt/create_prompt.py (hunk 1 above)
Comment (mamadoudicko) [reply backend/core/repository/prompt/create_prompt.py]: Ambigram_Say_Yes_radial_pattern_rainbow_color_-_rotation_animation

File: backend/core/models/databases/supabase/prompts.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+from models.prompt import Prompt
+from pydantic import BaseModel
+
+
+class CreatePromptProperties(BaseModel):
+    """Properties that can be received on prompt creation"""
+
+    title: str
+    content: str
+    status: str = "private"
```
Comment (gozineb) [inline backend/core/models/databases/supabase/prompts.py:15]: use an enum

File: backend/core/models/databases/supabase/prompts.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    content: Optional[str]
+    status: Optional[str]
+
+
+class Prompts(Repository):
+    def __init__(self, supabase_client):
+        self.db = supabase_client
+
+    def create_prompt(self, prompt: CreatePromptProperties) -> Prompt:
+        """Create a prompt by id"""
```
Comment (gozineb) [inline backend/core/models/databases/supabase/prompts.py:31]: by nada

**Review by gozineb — APPROVED** [review]

**Pull request merged**